File: packages/ios/Sources/Components/SegmentedControl/SegmentedControl.figma.swift
import SwiftUI
import Figma

struct KozmosSegmentedControlConnect: FigmaConnect {
    let component = KozmosSegmentedControl.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=309-5165"

    @FigmaString("Item 1 Text")
    var item1Text: String = "Overview"

    @FigmaString("Item 2 Text")
    var item2Text: String = "Details"

    @FigmaString("Item 3 Text")
    var item3Text: String = "Activity"

    @FigmaEnum(
        "Size",
        mapping: [
            "Small": KozmosSegmentedControlSize.sm,
            "Default": KozmosSegmentedControlSize.`default`,
            "Large": KozmosSegmentedControlSize.lg
        ]
    )
    var size: KozmosSegmentedControlSize = .default

    @FigmaEnum(
        "Active",
        mapping: [
            "One": 0,
            "Two": 1,
            "Three": 2
        ]
    )
    var activeIndex: Int = 0

    @FigmaEnum(
        "State",
        mapping: [
            "Default": "default",
            "Focus": "focus",
            "Disabled": "disabled",
            "Error": "error"
        ]
    )
    var state: String = "default"

    var body: some View {
        KozmosSegmentedControl(
            selection: .constant(self.activeIndex),
            items: [self.item1Text, self.item2Text, self.item3Text],
            size: self.size,
            disabled: self.state == "disabled"
        )
    }
}

File: packages/android/src/main/java/com/kozmos/components/SegmentedControl/SegmentedControl.figma.kt
package com.kozmos.components.segmentedcontrol

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=309-5165")
class KozmosSegmentedControlConnect {
    @FigmaProperty(FigmaType.Text, "Item 1 Text")
    val item1Text: String = "Overview"

    @FigmaProperty(FigmaType.Text, "Item 2 Text")
    val item2Text: String = "Details"

    @FigmaProperty(FigmaType.Text, "Item 3 Text")
    val item3Text: String = "Activity"

    @FigmaProperty(FigmaType.Enum, "Size")
    val size: SegmentedControlSize = Figma.mapping(
        "Small" to SegmentedControlSize.Sm,
        "Default" to SegmentedControlSize.Default,
        "Large" to SegmentedControlSize.Lg
    )

    @FigmaProperty(FigmaType.Enum, "Active")
    val activeIndex: Int = Figma.mapping(
        "One" to 0,
        "Two" to 1,
        "Three" to 2
    )

    @FigmaProperty(FigmaType.Enum, "State")
    val state: String = Figma.mapping(
        "Default" to "default",
        "Focus" to "focus",
        "Disabled" to "disabled",
        "Error" to "error"
    )

    @Composable
    fun ComponentExample() {
        KozmosSegmentedControl(
            options = listOf(item1Text, item2Text, item3Text),
            selectedIndex = activeIndex,
            onOptionSelected = {},
            size = size,
            enabled = state != "disabled"
        )
    }
}
